src/nvim/ex_docmd.c

```diff
@@ -6538,8 +6538,13 @@ void not_exiting(void)
 
 bool before_quit_autocmds(win_T *wp, bool quit_all, bool forceit)
 {
+  set_vim_var_nr(VV_CANCEL_EVENT, VV_FALSE);
   apply_autocmds(EVENT_QUITPRE, NULL, NULL, false, wp->w_buffer);
```

Comment:
perhaps the impelmenation would take the form of a check in `apply_autocmds()`, then all call sites would change to:
```suggestion
  if (!apply_autocmds(EVENT_QUITPRE, NULL, NULL, false, wp->w_buffer)) {
    return ...
  }
```

Reply:
I don't think I can do that... apply autocmds already returns a bool for if something ran.
wouldn't that conflict?

Reply:
in the case of QuitPre what that would translate to as I understand it, is if there wasn't any autocmds a window wouldn't close

Reply:
sure, my suggestion was just to present the concept. Implementation will be non-trivial.